examples/cpp/modes/vtol/include/mode.hpp

```diff
@@ -0,0 +1,139 @@
+/****************************************************************************
+ * Copyright (c) 2025 PX4 Development Team.
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+#pragma once
+
+#include <cmath>
+
+#include <px4_ros2/components/mode.hpp>
+#include <px4_ros2/odometry/global_position.hpp>
+#include <px4_ros2/control/setpoint_types/experimental/trajectory.hpp>
+#include <px4_ros2/control/setpoint_types/fixedwing_lateral_longitudinal.hpp>
+#include <px4_ros2/control/vtol.hpp>
+#include <px4_ros2/utils/geometry.hpp>
+
+#include <Eigen/Eigen>
+
+#include <rclcpp/rclcpp.hpp>
+
+using namespace std::chrono_literals; // NOLINT
+using namespace px4_ros2;
+
+static const std::string kName = "My VTOL Mode";
+
+class FlightModeTest : public px4_ros2::ModeBase
+{
+public:
+  explicit FlightModeTest(rclcpp::Node & node)
+  : ModeBase(node, kName)
+  { 
+    _vtol = std::make_shared<px4_ros2::VTOL>(*this);
+    _trajectory_setpoint = std::make_shared<px4_ros2::TrajectorySetpointType>(*this); 
+    _fixed_wing_setpoint = std::make_shared<px4_ros2::FwLateralLongitudinalSetpointType>(*this); 
+
+    _vehicle_global_position = std::make_shared<px4_ros2::OdometryGlobalPosition>(*this);
+  }
+
+  void onActivate() override 
+  {
+   VTOL::State initial_vtol_state = _vtol->get_current_state(); 
+
+    if(initial_vtol_state == VTOL::State::FIXED_WING){
+      _control_state = State::GoingNorth; 
+      std::cout << "VTOL Mode Activated. Flying North, climbing to 600m above sea level." << std::endl; 
+    } else if(initial_vtol_state == VTOL::State::UNDEFINED){
+      throw std::runtime_error("VTOL state undefined.");
+    } else{
+      _control_state = State::TransitionToFixedWing; 
+      std::cout << "VTOL Mode Activated. Transitioning to Fixed-Wing." << std::endl; 
```

Comment:
`RCLCPP_INFO(node().get_logger(), "VTOL Mode Activated. Transitioning to Fixed-Wing");`
Also below